pkg/controller/plan/migration.go

```diff
@@ -706,11 +709,34 @@ func (r *Migration) execute(vm *plan.VMStatus) (err error) {
 
 		switch vm.Phase {
```

Comment:
If you add the `WaitFor...` phases to the warm migration itinerary on line 103, you can replace this case with `vm.Phase = r.next(vm.Phase)`.

For example:

```
{Name: CreateInitialSnapshot},
{Name: WaitForInitialSnapshot},
{Name: CreateDataVolumes},
```

Reply:
Okay, I think I have flattened this one and the next one the way you wanted.